cli/azd/pkg/containerapps/container_app.go

```diff
@@ -127,6 +128,7 @@ const apiVersionKey = "api-version"
 
 var persistCustomDomainsFeature = alpha.MustFeatureKey("aca.persistDomains")
 var persistIngressSessionAffinity = alpha.MustFeatureKey("aca.persistIngressSessionAffinity")
+var persistCors = alpha.MustFeatureKey("aca.persistCors")
```

Comment:
We're going to end up with like 50 of these 😄

Reply:
Hopefully we will have the AppHost producing the `ACA definition` soon and get rid of all of these.

Reply:
Maybe the alpha feature should be something like: "give us the paths of keys that you want preserved" instead of adding a switch per property?  That would be a little more generalized and would at least force folks to think about the underlying container app object we are building.

Reply:
I'm not as familiar with many of the Aspire use cases - but why do we need to explicitly call out parts of the configuration vs taking the snapshot of the current version, then layer the yaml manifest config to form the final container app spec to update?

Reply:
When I mentioned snapshot I mean the call to getContainerApp that is already happening during deployment.  Then you additional have the yaml manifest that needs to get merged in.

Reply:
getContainerApp is not part of deployment.  We call it only if we want to persist something:

![image](https://github.com/user-attachments/assets/73d450f3-5335-48cf-8dac-43257a140133)

Reply:
I guess I'm not fully understanding the use case.  Why wouldn't we persist all of the configuration data that was set during the app provisioning?

Reply:
What we're doing here is ignoring IaC as the source-of-truth to allow users to overcome a current bottleneck in the design (i.e., ACA resource definition not being full representable at the AppHost layer):
- Bicep/IaC is usually the source-of-truth of the Azure resource definition.
- But here, we are saying "let's allow the user to tweak this value in Azure, and we will just ignore Bicep when we reprovision"

> "give us the paths of keys that you want preserved" instead of adding a switch per property

I think this would be most flexible. Can we make a `config.json` property for a `.JSONPath` to ignore?

```json
"aspire.containerapp.unmanaged.properties": [
   "properties.configuration.ingress.corsPolicy",
   "properties.configuration.ingress.stickySessions"
]
```

We would just need better docs to show how to do all these things. We could still make the alpha feature trigger such a thing underneath, but it'd perhaps make the code churn easier?

Reply:
Thanks for providing more context here @weikanglim.  I'm just not convinced that ignoring IaC as the source of truth is the right choice here.  

I'm all for users experimenting with configuration in the Azure Portal, but if the user wants to keep those changes they should go back and make the appropriate tweaks in their IaC  / or manifest, etc.

Is the root cause here that the current AppHost layer does not support all the knobs required to set/persist these types of settings?

Reply:
In case it wasn't clear from the previous message -- yes, this is a point-in-time thing. Aspire will eventually surface these knobs, but it requires modeling it at the AppHost layer, which I could imagine takes time.